File: Platforms/Bioma/BiomaSurfacePartonSoilSWATHourlyPartonC/src/cs/BiomaSurfacePartonSoilSWATHourlyPartonC/SurfacePartonSoilSWATHourlyPartonCExogenous.cs
using System;
using System.Collections.Generic;

public class SurfacePartonSoilSWATHourlyPartonCExogenous 
{
    private double _DayLength;
    private double _AirTemperatureMinimum;
    private double _GlobalSolarRadiation;
    private double _AirTemperatureMaximum;
    private double _AirTemperatureAnnualAverage;
    private double _HourOfSunset;
    private double _HourOfSunrise;
    
        public SurfacePartonSoilSWATHourlyPartonCExogenous() { }
    
    
    public SurfacePartonSoilSWATHourlyPartonCExogenous(SurfacePartonSoilSWATHourlyPartonCExogenous toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    _DayLength = toCopy._DayLength;
    _AirTemperatureMinimum = toCopy._AirTemperatureMinimum;
    _GlobalSolarRadiation = toCopy._GlobalSolarRadiation;
    _AirTemperatureMaximum = toCopy._AirTemperatureMaximum;
    _AirTemperatureAnnualAverage = toCopy._AirTemperatureAnnualAverage;
    _HourOfSunset = toCopy._HourOfSunset;
    _HourOfSunrise = toCopy._HourOfSunrise;
    }
    }
    public double DayLength
        {
            get { return this._DayLength; }
            set { this._DayLength= value; } 
        }
    public double AirTemperatureMinimum
        {
            get { return this._AirTemperatureMinimum; }
            set { this._AirTemperatureMinimum= value; } 
        }
    public double GlobalSolarRadiation
        {
            get { return this._GlobalSolarRadiation; }
            set { this._GlobalSolarRadiation= value; } 
        }
    public double AirTemperatureMaximum
        {
            get { return this._AirTemperatureMaximum; }
            set { this._AirTemperatureMaximum= value; } 
        }
    public double AirTemperatureAnnualAverage
        {
            get { return this._AirTemperatureAnnualAverage; }
            set { this._AirTemperatureAnnualAverage= value; } 
        }
    public double HourOfSunset
        {
            get { return this._HourOfSunset; }
            set { this._HourOfSunset= value; } 
        }
    public double HourOfSunrise
        {
            get { return this._HourOfSunrise; }
            set { this._HourOfSunrise= value; } 
        }
}

File: Platforms/Bioma/BiomaSurfacePartonSoilSWATHourlyPartonC/src/java/BiomaSurfacePartonSoilSWATHourlyPartonC/SurfacePartonSoilSWATHourlyPartonCExogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SurfacePartonSoilSWATHourlyPartonCExogenous
{
    private Double DayLength;
    private Double AirTemperatureMinimum;
    private Double GlobalSolarRadiation;
    private Double AirTemperatureMaximum;
    private Double AirTemperatureAnnualAverage;
    private Double HourOfSunset;
    private Double HourOfSunrise;
    
    public SurfacePartonSoilSWATHourlyPartonCExogenous() { }
    
    public SurfacePartonSoilSWATHourlyPartonCExogenous(SurfacePartonSoilSWATHourlyPartonCExogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.DayLength = toCopy.getDayLength();
            this.AirTemperatureMinimum = toCopy.getAirTemperatureMinimum();
            this.GlobalSolarRadiation = toCopy.getGlobalSolarRadiation();
            this.AirTemperatureMaximum = toCopy.getAirTemperatureMaximum();
            this.AirTemperatureAnnualAverage = toCopy.getAirTemperatureAnnualAverage();
            this.HourOfSunset = toCopy.getHourOfSunset();
            this.HourOfSunrise = toCopy.getHourOfSunrise();
        }
    }
    public Double getDayLength()
    { return DayLength; }

    public void setDayLength(Double _DayLength)
    { this.DayLength= _DayLength; } 
    
    public Double getAirTemperatureMinimum()
    { return AirTemperatureMinimum; }

    public void setAirTemperatureMinimum(Double _AirTemperatureMinimum)
    { this.AirTemperatureMinimum= _AirTemperatureMinimum; } 
    
    public Double getGlobalSolarRadiation()
    { return GlobalSolarRadiation; }

    public void setGlobalSolarRadiation(Double _GlobalSolarRadiation)
    { this.GlobalSolarRadiation= _GlobalSolarRadiation; } 
    
    public Double getAirTemperatureMaximum()
    { return AirTemperatureMaximum; }

    public void setAirTemperatureMaximum(Double _AirTemperatureMaximum)
    { this.AirTemperatureMaximum= _AirTemperatureMaximum; } 
    
    public Double getAirTemperatureAnnualAverage()
    { return AirTemperatureAnnualAverage; }

    public void setAirTemperatureAnnualAverage(Double _AirTemperatureAnnualAverage)
    { this.AirTemperatureAnnualAverage= _AirTemperatureAnnualAverage; } 
    
    public Double getHourOfSunset()
    { return HourOfSunset; }

    public void setHourOfSunset(Double _HourOfSunset)
    { this.HourOfSunset= _HourOfSunset; } 
    
    public Double getHourOfSunrise()
    { return HourOfSunrise; }

    public void setHourOfSunrise(Double _HourOfSunrise)
    { this.HourOfSunrise= _HourOfSunrise; } 
    
}